### PR TITLE
eng-169

### PR DESCRIPTION
**Pull request by Connorbelez**
### TL;DR

Added cash application and suspense resolution functionality to the cash ledger system with comprehensive test coverage.

### What changed?

Added two new integration functions to handle cash operations:

- `postCashApplication()` - Applies cash from UNAPPLIED_CASH or SUSPENSE accounts to borrower obligations, creating CASH_APPLIED or SUSPENSE_ESCALATED journal entries
- `postSuspenseResolution()` - Resolves suspense items through three methods: matching to obligations, writing off balances, or requesting refunds

Added corresponding admin mutations:

- `applyCashToObligation` - Validates inputs and calls postCashApplication with audit logging
- `resolveSuspenseItem` - Handles suspense resolution with different resolution types and audit logging

Created comprehensive test suites covering successful operations, edge cases, validation errors, and mutation-level functionality.

### How to test?

Run the test suites:
```bash
bun test convex/payments/cashLedger/__tests__/cashApplication.test.ts
bun test convex/payments/cashLedger/__tests__/suspenseResolution.test.ts
```

Test the mutations through the admin dashboard or API calls with the required parameters (sourceAccountId, targetObligationId, amount, reason, idempotencyKey).

### Why make this change?

This implements core cash management functionality needed for loan servicing operations. It allows administrators to properly apply received payments to borrower obligations and resolve suspense items that couldn't be automatically matched, ensuring accurate accounting and cash flow tracking.

## Summary by CodeRabbit

* **New Features**
  * Added admin capabilities to apply unapplied cash to borrower obligations with support for full and partial applications.
  * Added suspense resolution functionality supporting match, write-off, and refund operations.

* **Tests**
  * Added comprehensive test coverage for cash application and suspense resolution workflows.

### CODE REVIEW

File: convex/payments/cashLedger/mutations.ts [hunk 1]
```diff
@@ -233,3 +235,166 @@ export const writeOffObligationBalance = adminMutation
...
+				type: "refund",
+				amount: args.amount,
+				reason,
+			};
+		} else {
+			resolution = {
+				type: "write_off",
+				amount: args.amount,
+				reason,
+			};
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/payments/cashLedger/mutations.ts]: Fixed: `resolveSuspenseItem` builds refund/write_off resolution without embedding `reason`; audit and postings use the single trimmed `args.reason`.
